main.go

```diff
@@ -57,6 +61,19 @@ func save() {
 	}
 }
 
+func configureLogging(cfg *config.Config) {
+
+	if config.LogLevels.ContainsName(cfg.LogLevel) {
+		log.Printf("[main.configureLogging] Log Level is %s", cfg.LogLevel)
+		key, ok := config.LogLevels.GetKey(cfg.LogLevel)
+		if ok {
+			logr.SetThreshold(key)
+		}
+	} else {
+		log.Printf("[main.configureLogging] Unrecognised log level '%s', defaulting to INFO. Allowed values are [DEBUG, INFO, WARN, ERROR]", cfg.LogLevel)
+	}
+}
```

Comment:
I also just introduced the `logr.LevelFromString` function, which should help you with this.

The aforementioned function should return an error if the existing value isn't valid